File: sast-go/cases/accuracy/field_sensitive/one_dimensional_collection/numeric_index_state_solver/array_index_001_T/array_index_001_T.go

// evaluation information start
// real case = true
// evaluation item = 准确度->对象敏感与域敏感分析->区分一维字典/列表/数组的不同元素->索引值非数字的场景，能够区分不同索引上特定元素的状态（需要求解）
// scene introduction = 数组->数组索引->需求解
// level = 4
// bind_url = accuracy/field_sensitive/one_dimensional_collection/numeric_index_state_solver/array_index_001_T/array_index_001_T
// evaluation information end

package main
import "os/exec"

func array_index_001_T(__taint_src string) {
	var str = [3]string{"b", "c", __taint_src}
	__taint_sink(str[1+1])
}

func __taint_sink(o interface{}) {
	_ = exec.Command("sh", "-c", o.(string)).Run()
	}

func main() {
    __taint_src := "taint_src_value"
    array_index_001_T(__taint_src)
}

File: sast-js/case/completeness/single_app_tracing/datatype/array/array_index_001_T.js
// evaluation information start
// real case = true
// evaluation item = 完整度->单应用跟踪完整度->数据类型和结构->数组
// scene introduction = 数组索引读取
// level = 2
// bind_url = completeness/single_app_tracing/datatype/array/array_index_001_T
// date = 2025-12-16 09:24:29
// evaluation information end
const { execSync } = require("child_process");

function array_index_001_T(__taint_src) {
  // 场景特点：通过索引访问数组中的污点元素
  let arr = ["a", __taint_src, "c"];
  __taint_sink(arr[1]);
}

function __taint_sink(o) {
  execSync(o);
}

const taint_src = "taint_src_value";
array_index_001_T(taint_src);
